src/ILLink.Shared/DataFlow/StructStack.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ILLink.Shared.DataFlow
+{
+	public struct StructStack<TValue> : IEquatable<StructStack<TValue>>
+		where TValue : IEquatable<TValue>
+	{
+		private Stack<TValue>? Stack;
+
+		private readonly int _count = 0;
+
+		public StructStack ()
+		{
+		}
+
+		public StructStack (StructStack<TValue> newStack)
+		{
+			Stack = newStack.Stack == null ? null : new Stack<TValue> (newStack.Stack);
+		}
+
+		public StructStack (int count)
+		{
+			_count = count;
+		}
+
+		public int Count => Stack == null ? 0 : Stack.Count;
+
+		public bool Equals (StructStack<TValue> other)
+		{
+			if (this.Count != other.Count)
+				return false;
```

Comment:
This looks like it's a real count, not capacity... but then Push doesn't update it??

Reply:
This is the actual stack count, updated during the push. Changed the naming.